pkg/apis/extensions/validation/validation.go

```diff
@@ -51,38 +51,35 @@ func ValidateHorizontalPodAutoscalerName(name string, prefix bool) (bool, string
 	return apivalidation.ValidateReplicationControllerName(name, prefix)
 }
 
-func validateResourceConsumption(consumption *extensions.ResourceConsumption, fieldName string) errs.ValidationErrorList {
+func validateMetricUtilizations(utilizations []extensions.MetricUtilization, fieldName string) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	resource := consumption.Resource.String()
-	if resource != string(api.ResourceMemory) && resource != string(api.ResourceCPU) {
-		allErrs = append(allErrs, errs.NewFieldInvalid(fieldName+".resource", resource, "resource not supported by autoscaler"))
-	}
-	quantity := consumption.Quantity.Value()
-	if quantity < 0 {
-		allErrs = append(allErrs, errs.NewFieldInvalid(fieldName+".quantity", quantity, "must be non-negative"))
+	if len(utilizations) != 1 {
+		allErrs = append(allErrs, errs.NewFieldInvalid(fieldName, len(utilizations), "must containe exactly 1 element"))
+	} else {
+		name := utilizations[0].Name
+		if string(name) != string(extensions.MetricCPUUsage) && string(name) != string(extensions.MetricMemoryUsage) {
+			allErrs = append(allErrs, errs.NewFieldInvalid(fieldName+".name", name, "metric not supported by autoscaler"))
+		}
+		utilization := utilizations[0].Utilization.Value()
+		if utilization < 0 {
+			allErrs = append(allErrs, errs.NewFieldInvalid(fieldName+".utilization", utilization, isNegativeErrorMsg))
+		}
 	}
 	return allErrs
 }
 
 func validateHorizontalPodAutoscalerSpec(autoscaler extensions.HorizontalPodAutoscalerSpec) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if autoscaler.MinReplicas < 0 {
+	if autoscaler.MinReplicas != nil && *autoscaler.MinReplicas < 0 {
```

Comment:
Does MinReplicas == 0 make sense? Do you imagine using that to disable autoscaling, for instance? If it were to scale to 0, the autoscaler would currently have no signal to scale up.